lib/cloudregister/registerutils.py

```diff
@@ -601,6 +602,45 @@ def set_registry_auth_token(registry_fqdn, username, password):
     return updated
 
 
+# ----------------------------------------------------------------------------
+def clean_registry_setup():
+    """Remove the data previously set to make the registry work."""
+    remove_auth_entry()
+
+
+# ----------------------------------------------------------------------------
+def remove_auth_entry():
+    """Remove the auth token from the config json file."""
+    config_json, unset_failed = get_registry_credentials(False)
+    if config_json:
+        # we could open the credentials file
+        # and it is not empty
+        # unset the registry credentials
+        entry = None
+        smt = get_smt_from_store(__get_registered_smt_file_path())
+        if smt and smt.get_registry_FQDN():
+            logging.info(
+                'Unsetting the auth entry for %s' % smt.get_registry_FQDN()
+            )
+            entry = smt.get_registry_FQDN()
+        else:
+            logging.info('Unsetting the auth entry based on the token')
+            auth_token = __generate_registry_auth_token()
+            entries = config_json.get('auths', {}).items()
+            entry = [entry for entry, auth in entries if auth == auth_token]
+            entry = ''.join(entry)
+
+        config_json.get('auths', {}).pop(entry, {})
+        logging.info('Registry auth entry unset or not present on the file')
```

Comment:
We should check if our content is the only content in the file and if it is simply `unlink` the file.

Reply:
I would also delete the file instead of changing the content. The other logic for creating the credentials is also based on the presence of the file and iirc the logic above could leave a valid file with empty auths which confuses the creation of a new credentials file

Reply:
if the file has other content, I would not remove the file

The file gets removed when
- empty content -> `{}`
- not relevant content -> `{"auths": {}`
- only our entry -> `{"auths": {our_fqdn: our_auth_token}`

otherwise, files does not get removed